includes/discovery/vlans/q-bridge-mib.inc.php

```diff
@@ -1,26 +1,45 @@
 <?php
+require_once 'includes/discovery/vlans/vlan_functions.inc.php';
 
-echo 'IEEE8021-Q-BRIDGE-MIB VLANs : ';
+echo 'IEEE8021-Q-BRIDGE-MIB VLANs: ';
 
 $vlanversion = snmp_get($device, 'dot1qVlanVersionNumber.0', '-Oqv', 'IEEE8021-Q-BRIDGE-MIB');
 
 if ($vlanversion == 'version1' || $vlanversion == '2') {
-    echo "VLAN $vlanversion ";
+    echo "ver $vlanversion ";
 
+    $qbridge_data = array();
     $vtpdomain_id = '1';
-    $vlans        = snmpwalk_cache_oid($device, 'dot1qVlanStaticName', array(), 'IEEE8021-Q-BRIDGE-MIB');
+    $vlans        = snmpwalk_cache_oid($device, 'dot1qVlanStaticName', array(), 'Q-BRIDGE-MIB');
+    $tagoruntag   = snmpwalk_cache_oid($device, 'dot1qVlanCurrentEgressPorts', array(), 'Q-BRIDGE-MIB', null, '-OQUs --hexOutputLength=0');
+    $untag        = snmpwalk_cache_oid($device, 'dot1qVlanCurrentUntaggedPorts', array(), 'Q-BRIDGE-MIB', null, '-OQUs --hexOutputLength=0');
+    $base_indexes = snmpwalk_cache_oid($device, 'dot1dBasePortIfIndex', array(), 'BRIDGE-MIB');
 
     foreach ($vlans as $vlan_id => $vlan) {
-        echo " $vlan_id";
+        d_echo(" $vlan_id");
         if (is_array($vlans_db[$vtpdomain_id][$vlan_id])) {
             echo '.';
         } else {
-            dbInsert(array('device_id' => $device['device_id'], 'vlan_domain' => $vtpdomain_id, 'vlan_vlan' => $vlan_id, 'vlan_name' => $vlan['dot1qVlanStaticName'], 'vlan_type' => array('NULL')), 'vlans');
+            dbInsert(array(
+                'device_id' => $device['device_id'],
+                'vlan_domain' => $vtpdomain_id,
+                'vlan_vlan' => $vlan_id,
+                'vlan_name' => $vlan['dot1qVlanStaticName'],
+                'vlan_type' => array('NULL')
+            ), 'vlans');
             echo '+';
         }
 
         $device['vlans'][$vtpdomain_id][$vlan_id] = $vlan_id;
+
+        $id = "0.$vlan_id";
+        $untagged_indexes = q_bridge_bits2indices($untag[$id]['dot1qVlanCurrentUntaggedPorts']);
+        $egress_indexes = q_bridge_bits2indices($tagoruntag[$id]['dot1qVlanCurrentEgressPorts']);
```

Comment:
@murrant: I noticed you got rid of the array for tagged ports here.  The existing code only added tagged VLANs to ports_vlans, and used ports.ifVlan to keep track of the untagged vlan on a port.  Do we want to change the semantics of ports_vlans?  Maybe add a column to keep track of whether the VLAN is tagged?